extra/src/main/java/aima/extra/probability/domain/FiniteDomain.java

```diff
@@ -0,0 +1,45 @@
+package aima.extra.probability.domain;
+
+import java.util.Set;
+
+/**
+ * A Domain over a countable/discrete and finite set of objects.
+ * 
+ * @author Ciaran O'Reilly
+ * 
+ */
+public interface FiniteDomain extends DiscreteDomain {
+	/**
```

Comment:
I was thinking the same of adding generics from the higher interface. It is unlikely that different object types would be used to formulate the domain.